File: 1759. Count Number of Homogenous Substrings/1759. Count Number of Homogenous Substrings.cs
public class Solution {
    public int CountHomogenous(string s) {
        int counter = 0, consecutive = 1;
        for (int i = 1; i <= s.Length; i++) {
            if(i < s.Length && s[i] == s[i - 1]){
                consecutive++;
            }else{
                counter = (int)((counter + (long)consecutive * (consecutive + 1) / 2) % 1000000007);
                consecutive = 1;
            }
        }
        return counter;
    }
}

File: 1759. Count Number of Homogenous Substrings/1759. Count Number of Homogenous Substrings.cpp
class Solution {
public:
    int countHomogenous(string s) {
        long long counter = 0;
        int start = 0;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] != s[start]) {
                int j = i - start;
                while (j > 0){
                    counter += j;
                    j--;
                }
                start = i;
            }
        }

        int j = s.size() - start;
        while(j > 0){
            counter += j;
            j--;
        }

        return counter % 1000000007;
    }
};